UploadDaemon/Configuration/GlobPatternList.cs

```diff
@@ -75,5 +75,17 @@ public Boolean Matches(string text)
         {
             return includeRegexes.Any(regex => regex.IsMatch(text)) && !excludeRegexes.Any(regex => regex.IsMatch(text));
         }
+
+        /// <inheritdoc/>
+        public override bool Equals(object obj)
+        {
+            return this.Describe().Equals((obj as GlobPatternList)?.Describe());
+        }
+
+        /// <inheritdoc/>
+        public override int GetHashCode()
+        {
+            return this.Describe().GetHashCode();
```

Comment:
same here, please use a proper hash code. Martin P showed me a cool trick to make hashcode short in C#. C.f. here https://github.com/cqse/teamscale-profiler-dotnet/blob/f3ece7745aae386c87ffbcd4dab3d8b9a2a019b1/UploadDaemon/Configuration/AzureFileStorage.cs